The actual usage of GPU memory is much larger than the file size of segment

**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
1. do a search with  `gpu_resource_config.cache_capacity: 6GB`, but the actual GPU memory-usage in node Eros exceeds 6GB
```
Tue Sep 22 11:11:43 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 435.21       Driver Version: 435.21       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce RTX 208...  Off  | 00000000:02:00.0 Off |                  N/A |
| 53%   58C    P2    97W / 250W |   9237MiB / 11019MiB |     84%      Default |
+-------------------------------+----------------------+----------------------+
|   1  GeForce RTX 208...  Off  | 00000000:03:00.0 Off |                  N/A |
| 54%   61C    P2   251W / 250W |   7229MiB / 11019MiB |     72%      Default |
+-------------------------------+----------------------+----------------------+
|   2  GeForce RTX 208...  Off  | 00000000:83:00.0 Off |                  N/A |
| 27%   31C    P8     1W / 250W |     10MiB / 11019MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   3  GeForce RTX 208...  Off  | 00000000:84:00.0 Off |                  N/A |
| 27%   33C    P8     4W / 250W |     10MiB / 11019MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0     18996      C   /var/lib/milvus/bin/milvus_server           9225MiB |
|    1     18996      C   /var/lib/milvus/bin/milvus_server           7217MiB |
+-----------------------------------------------------------------------------+
```
2. The error log is when no enough GPU memory to allocate in node Poseidon
```
2020-09-22 02:15:00,941][ERROR][ENGINE][OnLoad][taskloader_th] Search task encounter exception: Failed to load index file: Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 2124800000 bytes (error 2 out of memory)
[2020-09-22 02:15:13,256][ERROR][SERVER][TakeToExecute][reqsched_thread] Req failed with code: Error code(30117): out of memory:  : Failed to load index file: Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 2124800000 bytes (error 2 out of memory)
```
```
02:15:14  Error: <BaseException: (code=24, message=out of memory:  : Failed to load index file: Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 2124800000 bytes (error 2 out of memory))>
02:15:14  ERROR:milvus.client.grpc_handler:Error: <BaseException: (code=24, message=out of memory:  : Failed to load index file: Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 2124800000 bytes (error 2 out of memory))>
02:15:14  2020-09-21 18:15:13,266 - milvus_benchmark - ERROR - <BaseException: (code=24, message=out of memory:  : Failed to load index file: Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 2124800000 bytes (error 2 out of memory))>
02:15:14  ERROR:milvus_benchmark:<BaseException: (code=24, message=out of memory:  : Failed to load index file: Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 2124800000 bytes (error 2 out of memory))>
02:15:14  2020-09-21 18:15:13,267 - milvus_benchmark - ERROR - Traceback (most recent call last):
02:15:14    File "main.py", line 80, in queue_worker
02:15:14      runner.run(run_type, collection)
02:15:14    File "/home/jenkins/agent/workspace/milvus-benchmark-0.11.0/milvus_benchmark/k8s_runner.py", line 382, in run
02:15:14      res = self.do_query(milvus_instance, collection_name, vec_field_name, top_ks, nqs, run_count, search_param, filter_query=filter_query)
```
3. the size of file
```
zong@zong-System-Product-Name:/test/milvus/db_data_011/sift_50m_4000000_128_l2/db/collections/C_1/P_1$ du -sh *
512	1220.map
2.1G	S_1031
2.1G	S_1133
1.3G	S_1202
301M	S_1219
2.1G	S_125
2.2G	S_226
2.2G	S_325
2.1G	S_432
2.1G	S_543
2.2G	S_641
2.2G	S_745
2.2G	S_845
2.2G	S_946
```

**Steps/Code to reproduce behavior**
The collection size is 50,000,000 and the segment_row_limit is 4,000,000 , the dim is 128 ( float )
search yaml
```
search_performance:
  collections:
    -
      server:
        db_config.primary_path: /test/milvus/db_data_011/sift_50m_4000000_128_l2
        cache_config.cpu_cache_capacity: 32GB
        engine_config.use_blas_threshold: 0
        engine_config.gpu_search_threshold: 100
        gpu_resource_config.enable: true
        gpu_resource_config.cache_capacity: 6GB
        gpu_resource_config.search_resources:
          - gpu0
          - gpu1
        gpu_resource_config.build_index_resources:
          - gpu0
          - gpu1
        wal_enable: true
      collection_name: sift_50m_4000000_128_l2
      run_count:  50
      top_ks: [1000]
      nqs: [1000, 1200]
      search_params:
        -
          nprobe: 16
```

**Expected behavior**

**Method of installation**
- [ ] Docker/cpu
- [x] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Softwars conditions (OS, CPU, GPU, Memory)


- Milvus version (master or released version)
-0.11.0

**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
paste-file-content-here
```

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

--- a/core/src/scheduler/TaskTable.cpp
+++ b/core/src/scheduler/TaskTable.cpp
@@ -174,7 +174,7 @@ TaskTable::PickToLoad(uint64_t limit) {
         } else if (table_[index]->state == TaskTableItemState::LOADED) {
             cross = true;
             ++loaded_count;
-            if (loaded_count > 2) {
+            if (loaded_count >= 2) {
                 return std::vector<uint64_t>();
             }
         } else if (table_[index]->state == TaskTableItemState::START) {

--- a/core/src/scheduler/TaskTable.cpp
+++ b/core/src/scheduler/TaskTable.cpp
@@ -174,7 +174,7 @@ TaskTable::PickToLoad(uint64_t limit) {
         } else if (table_[index]->state == TaskTableItemState::LOADED) {
             cross = true;
             ++loaded_count;
-            if (loaded_count >= 2) {
+            if (loaded_count >= 1) {
                 return std::vector<uint64_t>();
             }
         } else if (table_[index]->state == TaskTableItemState::START) {

